### PR TITLE
Ocrvs 2532: Document Uploader issue fixed.

### PR DESCRIPTION
**Pull request by emon-swe-sust**
fileUploader
uploadFile_1

### CODE REVIEW

File: packages/client/src/components/form/DocumentUploadfield/SimpleDocumentUploader.tsx [hunk 1]
```diff
@@ -52,6 +52,7 @@ type IFullProps = {
   error?: string
   disableDeleteInPreview?: boolean
   onComplete: (files: IAttachmentValue | {}) => void
+  toggleFileUploading?: (isUploading: boolean) => void
```
Comment (rikukissa) [inline packages/client/src/components/form/DocumentUploadfield/SimpleDocumentUploader.tsx:55]: It's probably best if this follows the `onSomething` naming convention. `onUploadingStateChanged` works for example

File: packages/client/src/components/form/FormFieldGenerator.tsx [hunk 2]
```diff
@@ -195,6 +195,7 @@ type GeneratedInputFieldProps = {
   error: string
   draftData?: IFormData
   disabled?: boolean
+  toggleFileUploading?: (isUploading: boolean) => void
```
Comment (rikukissa) [inline packages/client/src/components/form/FormFieldGenerator.tsx:198]: Same comment about naming here

**Review by rikukissa — CHANGES_REQUESTED** [review]
A small comment on naming, but overall looks good 👍

File: packages/client/src/components/form/DocumentUploadfield/SimpleDocumentUploader.tsx [hunk 3]
```diff
@@ -125,16 +140,24 @@ class SimpleDocumentUploaderComponent extends React.Component<
...
+    const errorMessage = requiredErrorMessage || this.state.error || error || ''
 
     return (
       <>
         {description && <FieldDescription>{description}</FieldDescription>}
         <ErrorMessage>
-          {errorMessage && (
-            <ErrorText id="field-error">{errorMessage}</ErrorText>
+          {errorMessage && touched && (
+            <ErrorText id="field-error">{errorMessage as string}</ErrorText>
```
Comment (rikukissa) [inline packages/client/src/components/form/DocumentUploadfield/SimpleDocumentUploader.tsx:160]: Have you tested what happens when you submit the `requiredErrorMessage` parameter as a MessageDescriptor object? Wouldn't that cause an object being rendered here as a string?

Comment (emon-swe-sust) [reply packages/client/src/components/form/DocumentUploadfield/SimpleDocumentUploader.tsx:160]: Yeah, I tested. and worked fine. 

Screenshot from 2022-03-02 12-30-59

this is the message from requiredErrorMessage.

**Pull request merged**